tests/integration/test_integration_complete.py

```
@@ -0,0 +1,283 @@
+    
+    print("\n🔧 Testing Integration Components (Offline)")
+    print("-" * 50)
+    
+    try:
+        # Test analytics service
+        from app.services.internal_analytics_service import get_analytics_service
+        analytics = get_analytics_service()
+        print("✅ Internal analytics service loaded successfully")
+        
+        # Test source detection service import
+        from app.services.source_detection_service import SourceDetectionService
```

Ruff (0.11.9)

`app.services.source_detection_service.SourceDetectionService` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)